catalog/dags/maintenance/pr_review_reminders/pr_review_reminders.py

```diff
@@ -28,33 +28,35 @@ class Urgency:
     MEDIUM = Urgency("medium", 4)
     LOW = Urgency("low", 5)
 
+    @classmethod
+    def for_pr(cls, pr: dict) -> Urgency:
+        priority_labels = [
+            label["name"]
+            for label in pr["labels"]
+            if "priority" in label["name"].lower()
+        ]
+        if not priority_labels:
+            logger.error(f"Found unlabeled PR ({pr['html_url']}). Skipping!")
+            return None
+
+        priority_label = priority_labels[0]
+
+        if "critical" in priority_label:
+            return cls.CRITICAL
+        elif "high" in priority_label:
+            return cls.HIGH
+        elif "medium" in priority_label:
+            return cls.MEDIUM
+        elif "low" in priority_label:
+            return cls.LOW
```

Comment:
I was curious what this might look like using Python's [structural pattern matching syntax](https://peps.python.org/pep-0622/) - in case you're interested as well!

```suggestion
        match priority_label.split()[-1]:
            case "critical":
                return cls.CRITICAL
            case "high":
                return cls.HIGH
            case "medium":
                return cls.MEDIUM
            case "low":
                return cls.LOW
```